### ISSUE TITLE
FormulateForm v-model removes properties not used in a FormulateInput

### ISSUE THREAD

**Issue by gregorybolkenstijn** [wearebraid/vue-formulate#108]
**Describe the bug**
I'm using the `FormulateForm` component in combination with `v-model` like so:

`<FormulateForm v-model="formData">`

Using the `FormulateInput` components will update the formData object accordingly. Then if I use another method to set data on the formData object like `formData.foo = 'Hello World'` I can still see the formData object update correctly. There is no `FormulateInput` component that is tied to the `foo` property. If I now use any `FormulateInput` component, the formData object gets updated with the new values, but any values that were set via another method like stated above are cleared to null. If I check the Vue.js devtools I can see the internal `formulateValue` object doing exactly the same: `foo` gets set too 'Hello World', but gets cleared afterwards when a `FormulateInput` component is used.

Any ideas on why this is happening?

**Comment by justin-schroeder** [comment]
Hi @gregorybolkenstijn!

That's a pretty good description of how it works yes. This is actually how the `v-model` contract works. Either side can set the value but internally to Vue `v-model`s are just setting a value on any `change` event, so for a text field it's basically `model = event.target.value`. When its a simple primitive like a text input there would be no expectation that there's other data to get wiped out.

In our case since we're v-modeling a whole form you can think of it like `model = getAllFormvalues()` on every change event. That wipes out any non-form data you might sprinkle in. Fortunately, computed properties make it pretty easy to add-back in whatever additional properties you need:

```js
export default {
  data () {
    values: {}, // v-model this
    extra: {
      foo: 'some extra value'
    } // other fields
  },
  computed: {
    mergedValues () {
      return { ...this.values, ...this.extra }
    }
  }
}
```

In the above example, just set your extra properties on `extra`. Currently I'm going to mark this as wontfix since it's working as expected, but I'll leave it open for a bit to see if there are any other concerns around this edge case.

**Comment by gregorybolkenstijn** [comment]
Thanks for the explanation. I already had exactly what you proposed, but considered it a workaround. Would it be possible to ignore the properties on the model that aren't tied to an input controlled by vue-formulate? Now it's causing side-effects which are unclear where they're coming from when debugging.

**Comment by justin-schroeder** [comment]
Everything is possible! And after playing around with it a bit I think I agree with you that there's a strong case to be made for having it work this way. Here's a codepen doodle of native Vue, where a group of checkboxes don’t jack with unused values in the array:

https://codepen.io/justin-schroeder/pen/zYrqvbr

So I think I'll do a little more research on this to determine:

1. Do we _want_ it to work this way?
2. How do we actually implement it without breaking changes?

In the meantime I'll re-open this.

**Comment by gahabeen** [comment]
@gregorybolkenstijn The #164 should interest you ;)

As a quick workaround, you can use this custom `FormulateForm`.
```vue
<template>
  <form :class="classes" @submit.prevent="formSubmitted">
    <FormulateSchema v-if="schema" :schema="schema" />
    <FormulateErrors v-if="!hasFormErrorObservers" :context="formContext" />
    <slot />
  </form>
</template>

<script>
import { shallowEqualObjects } from "@braid/vue-formulate/src/libs/utils";
import FormulateForm from "@braid/vue-formulate/src/FormulateForm";

export default {
  watch: {
    formulateValue: {
      handler(values) {
        /**
         * Overrides the default behavior to properly share v-model updates
         */
        if (this.isVmodeled && values && typeof values === "object") {
          const keys = Array.from(
            new Set(Object.keys(values).concat(Object.keys(this.proxy)))
          );
          keys.forEach(field => {
            if (!shallowEqualObjects(values[field], this.proxy[field])) {
              this.setFieldValue(field, values[field]);
              if (
                this.registry.has(field) &&
                !shallowEqualObjects(
                  values[field],
                  this.registry.get(field).proxy
                )
              ) {
                this.registry.get(field).context.model = values[field];
              }
            }
          });
        }
      },
      deep: true
    }
  }
};
</script>

```

**Comment by justin-schroeder** [comment]
@gregorybolkenstijn I believe this is resolved in `release/2.5.0` since it includes a deep watcher for proxy updating on 2.5 👍
